### PR TITLE
Add git sha to instance status page

### PR DESCRIPTION
**Pull request by tiina303**
## Changes

This would have been useful for me to see today. Based of https://github.com/PostHog/posthog/pull/5726 + comments there.

Note that `with.context: .` is needed so we build from local checkout instead (see https://github.com/docker/build-push-action#usage)

Verified that this works:
1. Built the package in with these temporary changes: https://github.com/PostHog/posthog/pull/5843/commits/315b14dc4e9e43d8e0d05fbb9b9b93de76a278d9 & https://github.com/PostHog/posthog/pull/5843/commits/d7084dad39182e66187485f3ee878773a5926896, which created https://hub.docker.com/r/posthog/posthog/tags?page=1&ordering=last_updated&name=testing
2. Deployed that to my Digital Ocean deployment via `helm upgrade -i -f https://raw.githubusercontent.com/digitalocean/marketplace-kubernetes/master/stacks/posthog/values.yml posthog posthog/posthog --create-namespace --namespace posthog --set image.tag=testing`
3. Checked the /instance/status page

Screen Shot 2021-09-07 at 23 58 30

Note that the sha is odd and doesn't match anything because I'm running this on top of a pull_request trigger, see docs: https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit

Debug info in the check output
```
Run git log -n 1
commit a3f6a872eb16733eb9420e01933acefa321e65fc
Author: Tiina Turban <tiina303@gmail.com>
Date:   Tue Sep 7 22:06:45 2021 +0000

    Merge 4ebc15109c382bbb31e1509042d4753637ebbe9b into 37a355687d17761c60152866e4e671c510125826
GIT_SHA = "Undefined"
GIT_SHA = 'a3f6a872eb16733eb9420e01933acefa321e65fc'
```
Which makes me believe it will work fine on top of master/main.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious
- [ ] New/changed UI is decent on smartphones (viewport width around 360px)

### CODE REVIEW

**Comment by macobo** [comment]
Would this work for any self-hosted instance. I'd assume that .git directory isn't checked in nor git-core would be installed in the docker container?

File: posthog/utils.py [hunk 1]
```diff
@@ -187,6 +187,19 @@ def get_git_commit() -> Optional[str]:
         return None
 
 
+def get_git_sha() -> Optional[str]:
```
Comment (Twixes) [inline posthog/utils.py:190]: This is the same as `get_git_commit`, only without arg `--short`, so I'd suggest just modifying that existing function by changing its signature to `def get_git_commit(short: bool = True) -> Optional[str]:`

**Review by Twixes — COMMENTED** [review]
It wouldn't work as far as I understand @macobo. I think this comment thread in the PR that this one is based on is still applicable: https://github.com/PostHog/posthog/pull/5726#discussion_r695671048

**Comment by tiina303** [comment]
> It wouldn't work as far as I understand @macobo. I think this comment thread in the PR that this one is based on is still applicable: #5726 (comment)

Forgot about that 😅 Just checked that in the Digital Ocean Kubernetes deployment we don't have git in the web pod.

**Review by fuziontech — APPROVED** [review]
yeah this would be super helpful.

I also really like how you did this on the build step for the container - well done!

File: .github/workflows/docker-image-test.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
               uses: docker/build-push-action@v2
               with:
                   push: false
-                  tags: posthog/posthog:latest
+                  tags: posthog/posthog:testing
```
Comment (macobo) [inline .github/workflows/docker-image-test.yml:34]: Is this a correct change? 100% unrelated to everything and quite suspect.

File: posthog/gitsha.py [hunk 3]
```diff
@@ -0,0 +1 @@
+GIT_SHA = "Undefined"
```
Comment (macobo) [inline posthog/gitsha.py:1]: This is the wrong way to do this. We have settings.py for this + environment variables.

**Comment by macobo** [comment]
In general we have HELM_INSTALL_INFO env var already. Why not chunk the sha/tag used in there and keep the code a lot simpler?

File: .github/workflows/docker-image-test.yml (hunk 2 above)
Comment (tiina303) [reply .github/workflows/docker-image-test.yml:34]: Forgot about this, but yes it is safer this way, which is why I left it in. As Michael called out already for the cases when someone changes `push:true`, if they then push a broken package from their wip PR to `latest` it wouldn't be all that great as that can get picked up by our users.

File: posthog/gitsha.py (hunk 3 above)
Comment (tiina303) [reply posthog/gitsha.py:1]: Exactly as Michael calls out here as well it seemed risky to change a file that can be changed in various ways. Though I should have left a comment at the top of this file explaining that it gets overwritten by our git workflows to make things super clear.

**Comment by tiina303** [comment]
> In general we have HELM_INSTALL_INFO env var already. Why not chunk the sha/tag used in there and keep the code a lot simpler?

I opted for this because HELM_INSTALL_INFO isn't always available (e.g. postgres backed PostHog versions, our cloud, someone using kubectl directly)

**Pull request merged**